### ISSUE TITLE
Events bound multiple times

### ISSUE THREAD

**Issue by massic80** [sebastiansulinski/three-state-checkbox#5]
Hi,
I noticed that there is no verification on whether some checkboxes have been already initialized or not.
In case of dynamic pages, where new checkboxes may appear, one could attempt to reinitialize all the page, so the checkboxes which were already initialized are bound to a new event. The checkboxes selection seems working, but I didn't test the callbacks. Nevertheless, instability apart, the duplicated events would bring a useless overhead.

I would use jQuery's event namespaces and "off" the previous events prior to re binding them (one could have changed the callbacks): something like `$(this).off('change.triState').on('change.triState', '.' + settings.classMaster, function() {`

**Comment by sebastiansulinski** [comment]
Please feel free to create a pull request for this feature.

**Comment by massic80** [comment]
I am attempting to, but I noticed that, meanwhile, you switched the `$(this)` events to `$(document)`.
Shall I follow this approach? With previous solution, I could attach and detach events to the same container, while now it's hard to detect which event is attached for an instance and which one for another.
You could say: detach them all. Yes, but I usually prefer more granularity :)
Why did you _bubble up_ to `document`?

**Comment by sebastiansulinski** [comment]
I've converted to `$(document)` to make it work with dynamically added elements - without the need to re-bind the plugin [http://api.jquery.com/on/#direct-and-delegated-events](http://api.jquery.com/on/#direct-and-delegated-events)

**Comment by massic80** [comment]
Actually, I'm not conviced; attaching the events to the document should allow it to work with any newly-addes set of checkboxed, but it has two consequences:
- in general, jQuery plugins attach the events to the targeted object, while this doesn't. It's unexpected.
- if one attempts to append the event twice, it's appended two times on the document, hence you can't detach it and, above all, **it may work once for anytime he appended it** (I didn't dig in the commit enough, for being sure about this)

**Comment by massic80** [comment]
I re-thought about this and I experienced a problem with a live instance of the plugin today.
If you append the events to `document`, if whatever intercepts the events in the "bubbling column" nothing works
I decided to create a new pull request which re-sets `this` in place of `document`: any new dynamic checkbox within the `this` context will still work.
Possibly, users can still instantiate $(document).ssdThreeStateCheckbox() and get the same result of before.
I'm also _restricting_ button's `click` to `this`.
